File: Year-2023/LeetCode-September-challenge/IsSubsequence.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, j = 0;

        while(i < s.size() && j < t.size()){
            if(s[i] == t[j])i++;
            j++;
        }
        return i == s.size();
    }
};

int main(){
    string s = "abc", t = "ahbgdc";
    Solution *S = new Solution();
    cout<<std::boolalpha<<S->isSubsequence(s, t)<<endl;
    return 0;
}

File: Year-2023/LeetCode-September-challenge/IsSubsequence.js
var isSubsequence = function (s, t) {
    let i = 0, j = 0;

    while (i < s.length && j < t.length) {
        if (s[i] === t[j]) ++i;
        ++j;
    }
    return i === s.length;
};

let s = "abc", t = "ahbgdc";
console.log(isSubsequence(s, t));